### ISSUE TITLE
[no-unused-capturing-group] Doesn't handle `toArray`

### ISSUE THREAD

**Issue by boris-petrov** [ota-meshi/eslint-plugin-regexp#986]
**Information:**
- ESLint version: 10.5.0
- `eslint-plugin-regexp` version: 3.1.0

`no-unused-capturing-group` correctly handles `[...query.matchAll(/as(d)/)]` (i.e. it *doesn't* report an error) but it does not handle (i.e. it gives a false-positive warning) `query.matchAll(/as(d)/).toArray()` which is exactly the same thing.

**Comment by ota-meshi** [comment]
To make it easier for everyone to contribute, could you please share code examples and reproducible repositories?

**Comment by boris-petrov** [comment]
I'm sorry, I thought it was clear that the code I gave in the original post is an example. Here you go:

```js
console.log([...'asd'.matchAll(/as(d)/gu)]);
console.log('asd'.matchAll(/as(d)/gu).toArray());
```

The first example doesn't lead to a `no-unused-capturing-group` warning, the second does even though they are equivalent.
